bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirCommandType.java

```diff
@@ -0,0 +1,862 @@
+        public static final int REPLY_GET_STATES = 0xd6;
+        public static final int REQUEST_SET_STATES = 0xd7;
+        public static final int REQUEST_GET_ERRORS = 0xd9;
+        public static final int REPLY_GET_ERRORS = 0xda;
+        public static final int REQUEST_SET_RESETS = 0xdb;
+        public static final int REQUEST_GET_HOURS = 0xdd;
+        public static final int REPLY_GET_HOURS = 0xde;
+        public static final int REQUEST_GET_BYPASS = 0xdf;
+        public static final int REPLY_GET_BYPASS = 0xe0;
+        public static final int REQUEST_GET_PREHEATER = 0xe1;
+        public static final int REPLY_GET_PREHEATER = 0xe2;
+        public static final int REQUEST_GET_RF = 0xe5;
+        public static final int REPLY_GET_RF = 0xe6;
+        public static final int REQUEST_GET_EWT = 0xeb;
+        public static final int REPLY_GET_EWT = 0xec;
+        public static final int REQUEST_SET_EWT = 0xed;
+    }
+
+    /**
+     * @return command key
+     */
+    public String getKey() {
+        return key;
+    }
+
+    /**
+     * @return data type for this command key
+     */
+    public @Nullable ComfoAirDataType getDataType() {
+        try {
+            return data_type.newInstance();
+        } catch (Exception e) {
+            logger.debug("Creating new DataType went wrong ", e);
+        }
+        return null;
+    }
+
+    /**
+     * @return possible byte values
+     */
+    public int @Nullable [] getPossibleValues() {
+        return possible_values;
+    }
+
+    /**
+     * @return relevant byte position inside the response byte value array
+     */
+    public int getChangeDataPos() {
+        return change_data_pos;
+    }
+
+    /**
+     * @return generate a byte value sequence for the response stream
+     */
+    public int[] getChangeDataTemplate() {
+        int[] template = new int[change_data_size];
+        for (int i = 0; i < template.length; i++) {
+            template[i] = 0x00;
+        }
+        return template;
+    }
+
+    /**
+     * @return byte position inside the request byte value array
+     */
+    public int @Nullable [] getGetReplyDataPos() {
+        return read_reply_data_pos;
+    }
+
+    /**
+     * @return bit mask for the response byte value
+     */
+    public int getGetReplyDataBits() {
+        return read_reply_data_bits;
+    }
+
+    /**
+     * Get single read command to update item.
+     *
+     * @param key
+     *
+     * @return ComfoAirCommand identified by key
+     */
+    public static @Nullable ComfoAirCommand getReadCommand(String key) {
+        ComfoAirCommandType commandType = ComfoAirCommandType.getCommandTypeByKey(key);
+
+        if (commandType != null) {
+            Integer getCmd = commandType.read_command == 0 ? null : Integer.valueOf(commandType.read_command);
+            Integer replyCmd = Integer.valueOf(commandType.read_reply_command);
+
+            return new ComfoAirCommand(key, getCmd, replyCmd, new int[0], null, null);
```

Comment:
This logic is used a lot. Should it not be wrapped into the `ComfoAirCommand` constructor?

Reply:
done